pkg/service/manifest/storage/storage.go

```diff
@@ -237,3 +238,54 @@ func (ms *Storage) DeleteResponse(id string) error {
 	}
 	return nil
 }
+
+// ReviewApplication does the following:
+//  1. Updates the application status according to the approved parameter.
+//  2. Creates a Credential Response corresponding to the approved parameter and with the given reason.
+//  3. Marks the operation with id == opID as done, and sets operation.Response to the StoredResponse from the object
+//     creates in step 2.
+//
+// The operation and it's response (from 3) are returned.
+func (ms *Storage) ReviewApplication(id string, approved bool, reason string, opID string, response StoredResponse) (*StoredResponse, *opstorage.StoredOperation, error) {
+	// TODO: everything should be in a single Tx.
+	m := map[string]any{
+		"status": opsubmission.StatusDenied,
+		"reason": reason,
+	}
+	if approved {
```

Comment:
concurreny access risk?

Reply:
Yes! This I'm leaving that for a separate PR. I wanted to get a simple implementation out, and then fix the concurrent access issues.